implementations/rust/ockam/ockam_identity/src/secure_channel/packets.rs

```diff
@@ -0,0 +1,57 @@
+use crate::credential::Credential;
+use crate::secure_channel::key_exchange_with_payload::KeyExchangeWithPayload;
+use crate::{IdentitiesCreation, IdentitiesVault, Identity};
+use alloc::sync::Arc;
+use ockam_core::vault::Signature;
+use ockam_core::Message;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, Message)]
+pub struct FirstPacket {
+    //noise_xx: <- e
+    pub(super) key_exchange: Vec<u8>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Message)]
+pub struct SecondPacket {
+    //noise_xx: -> e, ee, s, es
+    pub(super) key_exchange_with_payload: KeyExchangeWithPayload<IdentityAndCredential>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Message)]
+pub struct ThirdPacket {
+    //noise_xx: -> s, se
+    pub(super) key_exchange_with_payload: KeyExchangeWithPayload<IdentityAndCredential>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Message)]
+pub(super) struct IdentityAndCredential {
```

Comment:
@davide-baldo  I'm a bit lost on this one,   how is the on-the-wire representation of this struct?   it's using cbor?

Reply:
@polvorin I think that this is using `bare` but since we are changing the protocols we might as well bite the bullet and use `cbor` now